src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -147,14 +147,16 @@ runLanguageServer options userHandlers onInitialConfig onConfigChange getIdeStat
                               Left e  -> sendFunc $ wrap $ ResponseMessage "2.0" (responseId _id) Nothing (Just e)
                               Right r -> sendFunc $ wrap $ ResponseMessage "2.0" (responseId _id) (Just r) Nothing
                     ResponseAndRequest x@RequestMessage{_id, _params} wrap wrapNewReq act ->
-                        checkCancelled ide clearReqId waitForCancel lspFuncs wrap act x _id _params $
+                        checkCancelled ide clearReqId waitForCancel lspFuncs wrap act x _id _params $ do
```

Comment:
What's this `do` doing? And is it in the reader monad?

Reply:
we now have two case statements one after each other, so need to sequence them.  I have no idea what Monad it is in, I just made GHC happy :)

If someone wants to propose an alternative implementation for this I am happy to use it. Main thing is to populate either the error or result portion of the reply message.

Reply:
Both case statements are nested under a lambda which already has a `do` on it. This do only seems to wrap the function.

Reply:
Yeah that `do` doesn’t look right to me. Are you sure we cannot simply remove it?